### PR TITLE
port arm template to bicep

### PR DESCRIPTION
**Pull request by stishkin**
Bicep offers us:
- easier to write and understand Azure Deployment templates
- better tooling

### CODE REVIEW

**Comment by stishkin** [comment]
This is a first PR to enable splitting template into multiple files/modules which in turn enables a path to doing Azure Function slots

File: docs/how-to/add-edit-workbook.md [hunk 1]
```diff
@@ -38,7 +38,7 @@ While viewing an open workbook instance:
 Each workbook is stored as a serialized JSON string value in
 `deployments/workbook-data.json`.
 
-The serialized workbook data will be referenced in `azuredeploy.json` using the
+The serialized workbook data will be referenced in `azuredeploy.json` or `azuredeploy.bicep` using the
```
Comment (chkeita) [inline docs/how-to/add-edit-workbook.md:41]: why keeping both. Cant we just delete azuredeploy.json ?

File: src/deployment/azuredeploy.bicep [hunk 2]
```diff
@@ -0,0 +1,677 @@
...
+  }
+  {
+    name: 'syslogDataSourcesCron'
+    syslogName: 'cron'
+    kind: 'LinuixSyslog'
+  }
+  {
+    name: 'syslogDataSourcesDaemon'
+    syslogName: 'daemon'
+    kind: 'LinuixSyslog'
```
Comment (tevoinea) [inline src/deployment/azuredeploy.bicep:248]: Typo?
```suggestion
    kind: 'LinuxSyslog'
```

File: src/deployment/deploy.py [hunk 3]
```diff
@@ -158,6 +163,13 @@ def __init__(
         self.admins = admins
         self.allowed_aad_tenants = allowed_aad_tenants
 
+        if arm_or_bicep_template:
```
[the remark this answers is not in the document]
Comment (stishkin) [reply src/deployment/deploy.py:166]: depends, if there are users outthere that augmented their deployment template and rely on that - then we break them

[the remark this answers is not in the document]
Comment (stishkin) [reply src/deployment/deploy.py:166]: Ether way in next PR I'll remove the ARM support, and keep bicep template only

[the remark this answers is not in the document]
Comment (stishkin) [reply src/deployment/deploy.py:166]: For now I'll keep both, so it's easier to test and compare to make sure that I did not break anything

**Review by chkeita — APPROVED** [review]

File: src/deployment/deploy.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (stishkin) [reply src/deployment/deploy.py:166]: i need this to be supported for this PR

and next PR I'll remove it

File: docs/how-to/add-edit-workbook.md (hunk 1 above)
Comment (stishkin) [reply docs/how-to/add-edit-workbook.md:41]: will remove in next pr

File: src/deployment/azuredeploy.bicep (hunk 2 above)
Comment (tevoinea) [reply src/deployment/azuredeploy.bicep]: There were a few other other ones in this same block, sorry I only highlighted one.

Comment (stishkin) [reply src/deployment/azuredeploy.bicep]: yup, fixed those just now

File: src/deployment/azuredeploy.bicep [hunk 4]
```diff
@@ -0,0 +1,677 @@
...
+}
+
+resource eventGridSystemTopics 'Microsoft.EventGrid/systemTopics@2021-12-01' = {
+  name: fuzz_blob_topic_name
+  location: location
+  properties: {
+    source: storageAccounts[storageAccountIndex].id
+    topicType: 'microsoft.storage.storageaccounts'
+  }
+  resource evetnSubscriptions 'eventSubscriptions' = {
```
Comment (mgreisen) [inline src/deployment/azuredeploy.bicep:485]: evetnSubscriptions -> eventSubscriptions

**Pull request merged**